src/azul/terraform.py

```diff
@@ -815,24 +815,42 @@ def tf_config(self, app_name):
         key = 'x-amazon-apigateway-minimum-compression-size'
         openapi_spec[key] = config.minimum_compression_size
         assert 'aws_api_gateway_gateway_response' not in resources, resources
-        openapi_spec['x-amazon-apigateway-gateway-responses'] = {
-            f'DEFAULT_{response_type}': {
-                'responseParameters': {
-                    # Static value response header parameters must be enclosed
-                    # within a pair of single quotes.
-                    #
-                    # https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html#mapping-response-parameters
-                    # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-gateway-responses.html
-                    #
-                    # Note that azul.strings.single_quote() is not used here
-                    # since API Gateway allows internal single quotes in the
-                    # value, which that function would prohibit.
-                    #
-                    f'gatewayresponse.header.{k}': f"'{v}'"
-                    for k, v in AzulChaliceApp.security_headers.items()
-                }
-            } for response_type in ['4XX', '5XX']
-        }
+        openapi_spec['x-amazon-apigateway-gateway-responses'] = (
+            # Static value response header parameters must be enclosed within a
+            # pair of single quotes.
+            #
+            # https://docs.aws.amazon.com/apigateway/latest/developerguide/request-response-data-mappings.html#mapping-response-parameters
+            # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-gateway-responses.html
+            #
+            # Note that azul.strings.single_quote() is not used here since API
+            # Gateway allows internal single quotes in the value, which that
+            # function would prohibit.
+            {
+                f'DEFAULT_{response_type}': {
+                    'responseParameters': {
+                        f'gatewayresponse.header.{k}': f"'{v}'"
+                        for k, v in AzulChaliceApp.security_headers.items()
+                    }
```

Comment:
Shouldn't there be a Retry-After for the 5XXs?

The innermost dict comprehension is identical in both cases. Try to rewrite this so it is less duplicative.

Reply:
> Shouldn't there be a Retry-After for the 5XXs?

The Lambda timeout returns a 504, so it was decided we'd add the Retry-After to those responses. https://github.com/DataBiosphere/azul/issues/6284#issuecomment-2203683398